.github/workflows/build.yml

```diff
@@ -36,6 +36,13 @@ jobs:
       cibuildwheel_ver: "2.12.1"
       force_update: false
 
+  start_linux_ec2_runner:
+    uses: ./.github/workflows/ec2_runner_jobs.yml
+    secrets: inherit
+    permissions: write-all
```

Comment:
Does it actually need this permission? This will be trusting any third-party action (e.g. `machulav/ec2-github-runner`) called to do *anything* to our repo. AFAI can see, it shouldn't need any repo-level permission.

Reply:
It needs the permission to register the runner.
As each runner is created and destroyed for each build, it needs this permission to register it as a self-hosted runner for GHA.